sdk/core/src/test/java/com/klaviyo/core/auth/KlaviyoAuthTokenManagerRefreshTest.kt

```
@@ -552,6 +552,344 @@ class KlaviyoAuthTokenManagerRefreshTest : BaseTest() {
+
+        assertEquals("eager fetch must not invoke observers", false, notified)
+    }
+
+    @Test
+    fun `observer throwing does not prevent other observers`() = runTest(dispatcher) {
+        val jwt = makeJwt()
+        val provider = CountingSuccessProvider(jwt)
+        val manager = KlaviyoAuthTokenManager()
+
+        val receivedBySecond = mutableListOf<String>()
+        manager.onTokenRefresh { throw RuntimeException("boom") }
```

detekt (1.23.8)

[warning] RuntimeException is a too generic Exception. Prefer throwing specific exceptions that indicate a specific error case.

(detekt.exceptions.TooGenericExceptionThrown)